drivers/spi/spi_esp32_spim.c

```diff
@@ -103,6 +143,53 @@ static void IRAM_ATTR spi_esp32_isr(void *arg)
 }
 #endif
 
+static int spi_esp32_init_dma(const struct device *dev)
+{
+	const struct spi_esp32_config *cfg = dev->config;
+	struct spi_esp32_data *data = dev->data;
+	uint8_t channel_offset;
+
+	if (clock_control_on(cfg->clock_dev, (clock_control_subsys_t)cfg->dma_clk_src)) {
+		LOG_ERR("Could not enable DMA clock");
+		return -EIO;
+	}
+
+#ifdef CONFIG_SOC_ESP32C3
+	gdma_hal_init(&data->hal_gdma, 0);
+	gdma_ll_enable_clock(data->hal_gdma.dev, true);
+	gdma_ll_tx_reset_channel(data->hal_gdma.dev, cfg->dma_host);
+	gdma_ll_rx_reset_channel(data->hal_gdma.dev, cfg->dma_host);
+	gdma_ll_tx_connect_to_periph(data->hal_gdma.dev, cfg->dma_host, 0);
+	gdma_ll_rx_connect_to_periph(data->hal_gdma.dev, cfg->dma_host, 0);
```

Comment:
Question, is this prepared to the restrictions of GDMA in C3? It has only 3 channels which needs to be arbited and mux'ed with other perioherals.

Reply:
Not yet, by now this is the only driver that make use of GDMA, so the channel is hard-coded to the host index. When needed we will probably implement a proper driver for GDMA.